pkg/controller/node.go

```diff
@@ -55,13 +59,23 @@ func (c *Controller) nodeUpdate(prev, cur interface{}) {
 		return
 	}
 
-	if nodePrev.Spec.Unschedulable && util.MapContains(nodePrev.Labels, c.opConfig.NodeEOLLabel) ||
-		!nodeCur.Spec.Unschedulable || !util.MapContains(nodeCur.Labels, c.opConfig.NodeEOLLabel) {
+	// do nothing if the node should have already triggered an update or
+	// if only one of the label and the unschedulability criteria are met.
+	if !c.nodeIsReady(nodePrev) || c.nodeIsReady(nodeCur) {
 		return
 	}
+	c.movePodsOffNode(nodeCur)
+}
+
+func (c *Controller) nodeIsReady(node *v1.Node) bool {
+	return (!node.Spec.Unschedulable || util.MapContains(node.Labels, c.opConfig.NodeReadinessLabel) ||
```

Comment:
shouldn't the `!` outside of the parenthesis?

Reply:
Hm, the node is considered ready if  is not labeled as unschedulable or if it has a ready label or if it is a master. Otherwise (if it is unschedulable and not master and doesn't have a ready label) the node is not considered ready and we move pods off it.

Reply:
Alright, your statement makes sense and I also see your tests now. Thanks for clarification 👍